File: src/3174.clear-digits.3447/solution.ts
function clearDigits (s: string): string {
    const stack:string[] = [];
    for (let i = 0; i < s.length; i++) {
        const code = s.charCodeAt(i);
        if (stack.length === 0 || code < 48 || code > 57) {
            stack.push(s[i]);
        } else {
            stack.pop();
        }
    }
    return stack.join('');
}

File: src/3174.clear-digits.3447/solution.java
class Solution {
    public String clearDigits(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        char[] chars = s.toCharArray();
        for(char c : chars){
            if(Character.isDigit(c)){
                if(!stack.isEmpty() && !Character.isDigit(stack.peekLast())   ){
                    stack.pollLast();
                }else{
                    stack.addLast(c);
                }

            }else{
                stack.addLast(c);
            }
        }

        return stack.stream().map(String::valueOf).collect(Collectors.joining(""));

    }
}